### DISCUSSION TITLE
Export MarkdownParser and MarkdownSerialiser

### DISCUSSION THREAD

**Discussion by ShravanSunder** [aguingand/tiptap-markdown#25] [General]
**Is your feature request related to a problem? Please describe.**
I'm trying to get streamed markdown working with tiptap and this extension. 

**Describe the solution you'd like**
Could markdown parser be exported?  I'd like to use this however, i don't want to modify my current insertContent and insertContentAt.  It would be great to call it directly, manipulate the result and then insert it into tip tap

**Describe alternatives you've considered**
A lot of hacking

cc: @aguingand

**Comment by ShravanSunder** [comment]
I can create a PR if you'd like as well

**Comment by aguingand** [comment]
```js
editor.storage.markdown.parser.parse(content)
editor.storage.markdown.serializer.serialize(doc)
```
These are accessible so you can call it in a custom command. However it's not documented so it may change in a future version.

**Comment by ShravanSunder** [comment]
@aguingand I would like to use this extension's code and not have it change the behaviour of insertContent and inserContentAt.  Thats the reason i think the exports would be valuable.  Unless there other ways to accomplish this?

**Comment by aguingand** [comment]
I don't plan to export Parser & Serializer before v1.0. If you have a feature to add that can be passed as an option, feel free to make a PR. I will convert this to a discussion for now.

**Comment by aguingand** [comment]
Your issue can't be solved be overriding the command ?

```js
import { extensions } from '@tiptap/core';
import { Markdown } from 'tiptap-markdown';


Markdown.extend({
  addCommands() {
        const commands = extensions.Commands.config.addCommands();
        return {
            ...this.parent(),
            insertContentAt: (range, content, options) => (props) => {
                const parsed = props.editor.storage.markdown.parser.parse(content, { inline: true });
                 // do something here
                return commands.insertContentAt(
                    range,
                    parsed,
                    options
                )(props);
            },
        }
    },
});
```

**Comment by bgeils** [comment]
@aguingand is there a way to extend Markdown Extension to write a custom serializer... we have a custom Node type in TipTap and use `editor.storage.markdown.getMarkdown()` ... ideally we would write a small serialize function that gets used for that node type.
